Code/Framework/AzToolsFramework/AzToolsFramework/Script/LuaEditorSystemComponent.cpp

```diff
@@ -0,0 +1,236 @@
+
+            auto luaComponentAssetCreator = [&](const AZStd::string& fullSourceFolderNameInCallback, [[maybe_unused]] const AZ::Uuid& sourceUUID)
+            {
+                AZStd::string defaultScriptName = "NewComponent";
+
+                AZStd::string fullFilepath;
+                AZ::StringFunc::Path::ConstructFull(fullSourceFolderNameInCallback.c_str(),
+                    defaultScriptName.c_str(),
+                    LuaExtension,
+                    fullFilepath);
+
+                MakeFilenameUnique(fullSourceFolderNameInCallback, defaultScriptName, fullFilepath);
+
+                AZStd::string scriptBoilerplate = GenerateLuaComponentBoilerplate(AZ::IO::Path(fullFilepath).Stem().Native());
+
+                auto outcome = SaveLuaScriptFile(fullFilepath, scriptBoilerplate);
+                if (outcome.IsSuccess())
+                {
+                    AzToolsFramework::AssetBrowser::AssetBrowserFileCreationNotificationBus::Event(
+                        AzToolsFramework::AssetBrowser::AssetBrowserFileCreationNotifications::FileCreationNotificationBusId,
+                        &AzToolsFramework::AssetBrowser::AssetBrowserFileCreationNotifications::HandleAssetCreatedInEditor,
+                        fullFilepath,
+                        LuaComponentScriptBusId);
+                }
+                else
+                {
+                    AZ_Error(LogName, false, outcome.GetError().c_str());
+                }
+            };
+
+            creators.push_back({ "LuaComponent_creator", "Lua Component Script", QIcon(), luaComponentAssetCreator });
+        }
+
+        void LuaEditorSystemComponent::AddSourceFileOpeners(
+            const char* fullSourceFileName,
+            [[maybe_unused]] const AZ::Uuid& sourceUUID,
+            AzToolsFramework::AssetBrowser::SourceFileOpenerList& openers)
+        {
+            if (AZ::IO::Path(fullSourceFileName).Extension() == LuaExtension)
+            {
+                auto luaScriptOpener = [](const char* fullSourceFileNameInCallback, [[maybe_unused]] const AZ::Uuid&)
+                {
+                    AzToolsFramework::EditorRequestBus::Broadcast(
+                        &AzToolsFramework::EditorRequests::LaunchLuaEditor, fullSourceFileNameInCallback);
+                };
+
+                openers.push_back({ "O3DE_LUA_Editor", "Open in Open 3D Engine LUA Editor...", QIcon(), luaScriptOpener });
+            }
+        }
+
+        void LuaEditorSystemComponent::HandleInitialFilenameChange(const AZStd::string_view fullFilepath)
+        {
+            AZ::IO::Path filepath = AZ::IO::Path(fullFilepath);
+            if (filepath.Extension() == LuaExtension)
+            {
+                AZStd::string_view& filename = filepath.Stem().Native();
+                AZStd::string scriptBoilerplate = GenerateLuaComponentBoilerplate(filename);
+
+                auto outcome = SaveLuaScriptFile(fullFilepath, scriptBoilerplate);
```

Comment:
Nit: could be `const`

Reply:
Ooh, nice. I was thinking that these weren't `const` since they modify stuff, but technically they are since they don't modify the object state itself. Thanks!

Reply:
Did it anyway. Had a chat with Danilo about how `const` on local variables can improve code clarity and maintenance. I can see how it could also be confusing to astute readers though, if you forget to `const` something that could have been `const`. Seems like a small price to pay overall though.